carsons/carsons.py

```diff
@@ -183,3 +181,75 @@ def calculate_distance(positionᵢ, positionⱼ):
     def get_h(self, i):
         _, yᵢ = self.phase_positions[i]
         return yᵢ
+
+    @property
+    def impedance(self):
+        z_primitive = self.build_z_primitive()
+        z_abc = perform_kron_reduction(z_primitive)
+        return z_abc
+
+
+class ConcentricNeutralCarsonsEquations(CarsonsEquations):
+    def __init__(self, model, *args, **kwargs):
+        super().__init__(model)
+        self.neutral_strand_gmr = model.neutral_strand_gmr
+        self.neutral_strand_count = defaultdict(
+            lambda: None, model.neutral_strand_count)
+        self.neutral_strand_resistance = model.neutral_strand_resistance
+        self.radius = defaultdict(lambda: None, {
+            phase: (diameter_over_neutral -
+                    model.neutral_strand_diameter[phase]) / 2
+            for phase, diameter_over_neutral
+            in model.diameter_over_neutral.items()
+        })
+        self.phase_positions.update({
+            f"N{phase}": self.phase_positions[phase]
+            for phase in self.phase_positions.keys()
+        })
+        self.gmr.update({
+            phase: self.GMR_cn(phase)
+            for phase in model.diameter_over_neutral.keys()
+        })
+        self.r.update({
+            phase: resistance / model.neutral_strand_count[phase]
+            for phase, resistance in model.neutral_strand_resistance.items()
+        })
+        return
+
+    def compute_d(self, i, j):
+        I, J = set(i), set(j)
+        r = self.radius[i] or self.radius[j]
+        if I ^ J == set('N') and 'N' not in I & J:
+            return r
+
+        distance_ij = self.calculate_distance(self.phase_positions[i],
+                                              self.phase_positions[j])
+        if not I & J and 'N' in I ^ J:
+            # approximate by modelling the concentric neutral cables as one
+            # equivalent conductor directly above the phase conductor
+            return (distance_ij**2 + r**2) ** 0.5
+        else:
+            return distance_ij
+
+    def compute_P(self, i, j, number_of_terms=1):
```

Comment:
Should `number_of_terms` be passed through here?

Reply:
seems like it's exactly the same as in `CarsonsEquations` class... thinking of dropping it here

Reply:
dropped it here 53fa95f